extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -299,13 +348,130 @@ export class AmpStoryQuiz extends AMP.BaseElement {
    * @private
    */
   handleOptionSelection_(optionEl) {
-    this.triggerAnalytics_(optionEl);
+    this.responseDataPromise_.then(() => {
+      this.triggerAnalytics_(optionEl);
 
-    this.mutateElement(() => {
-      optionEl.classList.add('i-amphtml-story-quiz-option-selected');
-      this.quizEl_.classList.add('i-amphtml-story-quiz-post-selection');
+      this.mutateElement(() => {
+        optionEl.classList.add('i-amphtml-story-quiz-option-selected');
+        this.quizEl_.classList.add('i-amphtml-story-quiz-post-selection');
+
+        this.hasUserSelection_ = true;
+      });
+
+      this.updateReactionData_(optionEl.optionIndex_);
+    });
+  }
+
+  /**
+   * Get the Reaction data from the datastore
+   *
+   * @return {Promise<JsonObject>}
+   * @private
+   */
+  retrieveReactionData_() {
+    return this.executeReactionRequest_()
+      .then(response => {
+        if (response.hasNoEndpoint) {
+          return;
+        }
+
+        this.handleSuccessfulDataRetrieval_(response);
+      })
+      .catch(error => {
+        dev().error(TAG, error);
+      });
+  }
+
+  /**
+   * Update the Reaction data in the datastore
+   *
+   * @param {number} reactionValue
+   * @private
+   */
+  updateReactionData_(reactionValue) {
+    this.executeReactionRequest_(reactionValue).catch(error => {
+      dev().error(TAG, error);
+    });
+  }
+
+  /**
+   * Executes a Reactions API call.
+   *
+   * @param {number} reactionValue
+   * @return {Promise<JsonObject>}
+   * @private
+   */
+  executeReactionRequest_(reactionValue) {
+    // TODO(jackbsteinberg): Add a default reactions endpoint.
+    if (!this.element.hasAttribute('endpoint')) {
+      return Promise.resolve({hasNoEndpoint: true});
+    }
+
+    let URL = this.element.getAttribute('endpoint');
+
+    const requestVars = {
```

Comment:
I think this should be:

```javascript
const requestVars = dict({
  ...
});
```

And subsequent properties should be accessed using the bracket syntax (`requestVars['reactionValue']`) instead of the dot syntax (`requestVars.reactionValue`).  Same for `requestOptions` below.  This is a signal to the Closure Compiler that you actually care about the name of the key here, otherwise during minification, you could end up with something like `a.b` instead of `a.reactionValue` (and then the server would not be able to understand the request)